plugins/modules/remote_management/redfish/ilo_redfish_command.py

```diff
@@ -0,0 +1,341 @@
+            type: dict
+            contains:
+                ret:
+                    description: Return True/False based on whether the operation was performed succesfully.
+                    type: bool
+                msg:
+                    description: Status of the operation performed on the iLO.
+                    type: dict
+    returned: always
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.community.general.plugins.module_utils.ilo_redfish_utils import (
+    iLORedfishUtils, iLO_certificate_login
+)
+from ansible.module_utils.common.text.converters import to_native
+
+# More will be added as module features are expanded
+CATEGORY_COMMANDS_ALL = {
+    "Systems": ["VerifyServiceBiosAttributes", "VerifyBiosAttributes", "VerifyLogicalDrives", "VerifyUefiBootOrder",
+                "VerifySpecifiedLogicalDrives", "CheckRebootStatus", "GetSpecifiedLogicalDrives"]
+}
+
+
+def main():
+    result = {}
+    module = AnsibleModule(
+        argument_spec=dict(
+            category=dict(required=True),
+            command=dict(required=True, type='list', elements='str'),
+            baseuri=dict(required=True),
+            timeout=dict(type="int", default=60),
+            username=dict(),
+            password=dict(no_log=True),
+            auth_token=dict(no_log=True),
+            service_attributes=dict(type="dict"),
+            bios_attributes=dict(type="dict"),
+            raid_details=dict(type="list", elements='dict'),
+            uefi_boot_order=dict(type='list', elements='str'),
+            logical_drives_names=dict(type='list', elements='str'),
+            cert_file=dict(type="str"),
+            key_file=dict(type="str")
+        ),
+        required_together=[
+            ("username", "password"),
+            ("cert_file", "key_file")
+        ],
+        required_one_of=[
+            ("username", "auth_token", "cert_file"),
+        ],
+        mutually_exclusive=[
+            ("username", "auth_token", "cert_file"),
+        ],
+        supports_check_mode=False
+    )
+
+    category = module.params['category']
+    command_list = module.params['command']
+
+    # admin credentials used for authentication
+    creds = {'user': module.params['username'],
+             'pswd': module.params['password'],
+             'token': module.params['auth_token']}
+
+    timeout = module.params['timeout']
+
+    # Build root URI
+    root_uri = "https://" + module.params['baseuri']
+
+    if module.params["cert_file"]:
+        creds["token"] = iLO_certificate_login(root_uri, module, module.params["cert_file"], module.params["key_file"])
+
+    rf_utils = iLORedfishUtils(creds, root_uri, timeout, module)
+
+    # Check that Category is valid
+    if category not in CATEGORY_COMMANDS_ALL:
+        module.fail_json(msg=to_native(
+            "Invalid Category '%s'. Valid Categories = %s" % (category, list(CATEGORY_COMMANDS_ALL.keys()))))
```

Comment:
use `choices` in the definition of the `category` parameter instead.

Reply:
added choices

Reply:
Now that you did, this check here becomes redundant.